IPFS Desktop 0.9 released

hacdias opened the pull request:
This is a draft of a post highlighting the new features of IPFS Desktop and some other stuff. I'll be updating this really soon

License: MIT
Signed-off-by: Henrique Dias <hacdias@gmail.com>

momack2 commented on content/post/057-ipfs-desktop-0-9.md:
```diff
@@ -0,0 +1,65 @@
+author: Henrique Dias
+---
+
+IPFS Desktop has come a long way to being what it is nowadays. We are here to cellebrate and announce the [release of 0.9.0](https://github.com/ipfs-shipyard/ipfs-desktop/releases) and, with that, tell a bit about the story of this application and, of course, the new features.
```
```suggestion
IPFS Desktop has come a long way to reach its current form. We are here to celebrate and announce the [release of 0.9.0](https://github.com/ipfs-shipyard/ipfs-desktop/releases) and share the story of this application and, of course, all the exciting new features.
```

momack2 commented on content/post/057-ipfs-desktop-0-9.md:
```diff
@@ -0,0 +1,65 @@
+
+## Once upon a time...
+
+It was back in May 2015 when Juan Benet pushed a first so-called "Initial Commit" to start a project at the time named "ipfs-electron". The purpose of the app at the time was quite simple and that hasn't changed much with time: providing a long running daemon, alongside with a GUI for managing it.
```
```suggestion
Back in May 2015, Juan Benet pushed the first so-called "Initial Commit" to start a project at the time named "ipfs-electron". The purpose of the app was quite simple and that hasn't changed much with time: providing a long running daemon, alongside with a GUI for managing it.
```

momack2 commented on content/post/057-ipfs-desktop-0-9.md:
```diff
@@ -0,0 +1,65 @@
+
+### Browse the wild...
+
+With the recent update to the Web UI, we are now capable of navigating throug the whole IPFS system. Previously, you could only access your 'Files' - Mutable File System for those who enjoy technical terms. Now, you can navigate not only to any `/ipfs/` or `/ipns/` path, but also to your pins!
```
```suggestion
With the recent update to the Web UI, we are now capable of navigating through the whole IPFS system. Previously, you could only access your 'Files' (the contents of the [Mutable File System](https://docs.ipfs.io/guides/concepts/mfs/) for those who enjoy technical terms). Now, you can navigate not only to any `/ipfs/` or `/ipns/` path, but also any content [pinned](https://docs.ipfs.io/guides/concepts/pinning/) to your IPFS node!
```
I'd suggest adding links to the terminology you use so that beginners can better understand what we mean by these commands. A lot of nontechnical users read our blog - and this is a great onboarding opportunity for them!

momack2 commented on content/post/057-ipfs-desktop-0-9.md:
```diff
@@ -0,0 +1,65 @@
+
+### Use npm on ipfs
+
+In addition, we are introducing a new experiments section on the 'Settings' page which will allow you to enable new experimental features as they come in. To cellebrate this new section, we are adding [npm on ipfs](https://github.com/ipfs-shipyard/npm-on-ipfs) to it! This is a project that allows you to use `ipfs-npm` command to install your Node.js modules through the IPFS network!
```
```suggestion
In addition, we are introducing a new experiments section on the 'Settings' page which will allow you to enable new experimental features as they come in. To celebrate this new section, we are adding [npm on ipfs](https://github.com/ipfs-shipyard/npm-on-ipfs) to it! This is a project that allows you to use the `ipfs-npm` command to install your Node.js modules through the IPFS network!
```

momack2 commented on content/post/057-ipfs-desktop-0-9.md:
```diff
@@ -0,0 +1,65 @@
+
+![Experiments section](/057-ipfs-desktop-0-9/npm-ipfs.png)
+
+**Please note** that this feature is intended for those who have Node.js installed on their system. Moreover, the activation will only work if your setup does not require additional permissions for executing `npm install -g`.
```
is there anything folks can do if their setup DOES require additional permissions, or are they SOL?

hacdias replied:
On that case, we recommend them to execute the command themselves (`npm i -g npm-ipfs`). That's the message we show if it fails.

autonome replied:
Can you link Node.js to its install page?

momack2 commented on content/post/057-ipfs-desktop-0-9.md:
```diff
@@ -0,0 +1,65 @@
+
+### Add `ipfs` command line tools to your system
+
+Adding `ipfs` command line tools to your system is not a new feature and it was introduced in 0.8.0. Although, it's always worth noting that you are a click away of having the `ipfs` command available from your command line! Just go to Settings and enable "IPFS command line tools"!
```
```suggestion
Adding `ipfs` command line tools to your system was actually introduced in 0.8.0, but it's always worth noting that you are just a click away of having the `ipfs` command available from your command line! Just go to Settings and enable "IPFS command line tools"!
```

momack2 replied:
Another sentence or two about the importance of this (aka you don't have to worry about configuring your PATH variable or any confusing stuff to start using `ipfs` on the command line after installing desktop and enabling these tools!)

momack2 commented on content/post/057-ipfs-desktop-0-9.md:
```diff
@@ -0,0 +1,65 @@
+
+There are some other features that have existed for quite a while already, ...:
+
+- **Handle `ipfs://`, `ipns://` and `dweb:` links**. If you have IPFS Desktop installed and you click on [ipns://ipfs.io](ipns://ipfs.io), that request will go through the application and redirected your own gateway or to the public gateway if yours is not online.
```
```suggestion
- **Handle `ipfs://`, `ipns://` and `dweb:` links**. If you have IPFS Desktop installed and you click on [ipns://ipfs.io](ipns://ipfs.io), that request will go through be redirected to your own gateway via the app, or to the public gateway if yours is not online.
```

momack2 commented on content/post/057-ipfs-desktop-0-9.md:
```diff
@@ -0,0 +1,65 @@
+
+### And more...
+
+There are some other features that have existed for quite a while already, ...:
```
```suggestion
Other notable features from previous releases include:
```

momack2 commented on content/post/057-ipfs-desktop-0-9.md:
```diff
@@ -0,0 +1,65 @@
+
+# Let's talk about future...
+
+For the future of IPFS Desktop, we are now focusing much more on maintenance, bug fixing and improving the test suite to catch bugs before they go into production. Being this an Electron app, it is quite harder to test than some other types of apps. If you're interesting in knowing more, please read the issue [#1121](https://github.com/ipfs-shipyard/ipfs-desktop/issues/1121).
```
```suggestion
For the future of IPFS Desktop, we are now focusing much more on maintenance, bug fixing and improving the test suite to catch bugs before they go into production. Since this an Electron app, it is a bit harder to test than some other types of apps. If you're interesting in knowing more, please read the issue [#1121](https://github.com/ipfs-shipyard/ipfs-desktop/issues/1121).
```

momack2 commented on content/post/057-ipfs-desktop-0-9.md:
```diff
@@ -0,0 +1,65 @@
+
+# Don't forget...
+
+Don't forget to [install IPFS Desktop](https://github.com/ipfs-shipyard/ipfs-desktop/releases) today and be part of the decentralized Internet! We also recommend you to install our browser extension, [IPFS Companion](https://github.com/ipfs-shipyard/ipfs-companion/#install), which gives your browser superpowers.
```
This is great! Another sentence about each and how they work together would help put this in context as well

momack2 commented:
This is awesome!!! Great initiative to write this up. Made some suggestions and nits, but love the direction you're heading. =]

hacdias commented:
@momack2 thanks a lot for the review! I was just waiting to release some bug fixes to write a post and so here we are 😄

autonome commented on content/post/057-ipfs-desktop-0-9.md:
```diff
@@ -0,0 +1,65 @@
+
+![Evolution of IPFS Desktop](/057-ipfs-desktop-0-9/evolution.jpg)
+
+Regarding the application's name, it went from "ipfs-electron" to "IPFS Native Application", then to "IPFS Station", it being a much more fancier name. However, we changed it again to "IPFS Desktop" and that's where we are right now!
```
s/more//

autonome replied:
Would be neat to link to the latest name change and summarize the reasoning behind the decision, even if it's just because it's descriptive 😂

autonome commented on content/post/057-ipfs-desktop-0-9.md:
```diff
@@ -0,0 +1,65 @@
+
+Regarding the application's name, it went from "ipfs-electron" to "IPFS Native Application", then to "IPFS Station", it being a much more fancier name. However, we changed it again to "IPFS Desktop" and that's where we are right now!
+
+## IPFS Desktop 0.9.0
```
To flow more smoothly from the previous section, maybe "Where we are today: IPFS Desktop 0.9.0" or something like that?

autonome commented on content/post/057-ipfs-desktop-0-9.md:
```diff
@@ -0,0 +1,65 @@
+
+## IPFS Desktop 0.9.0
+
+The newest version of IPFS Desktop, 0.9.2 - yes, we already released two patch versions! - brings more stability and some new experimental features. Although it is not a completely revamped application, such as what happened with 0.7.0, I believe this version deserves an overview of the current (new) features and plans for the future.
```
s/(new)/(and new)/

?

autonome commented on content/post/057-ipfs-desktop-0-9.md:
```diff
@@ -0,0 +1,65 @@
+
+### Browse the wild...
+
+With the recent update to the Web UI, we are now capable of navigating through the whole IPFS system. Previously, you could only access your 'Files' (the contents of the [Mutable File System](https://docs.ipfs.io/guides/concepts/mfs/) for those who enjoy technical terms). Now, you can navigate not only to any `/ipfs/` or `/ipns/` path, but also any content [pinned](https://docs.ipfs.io/guides/con ...
```
Link "recent update to the Web UI" to that release announcement?

autonome replied:
I recommend removing "for those who enjoy technical terms".

Hm, should we link to the MFS guide, or to the new ProtoSchool tutorial...

autonome commented on content/post/057-ipfs-desktop-0-9.md:
```diff
@@ -0,0 +1,65 @@
+
+With the recent update to the Web UI, we are now capable of navigating through the whole IPFS system. Previously, you could only access your 'Files' (the contents of the [Mutable File System](https://docs.ipfs.io/guides/concepts/mfs/) for those who enjoy technical terms). Now, you can navigate not only to any `/ipfs/` or `/ipns/` path, but also any content [pinned](https://docs.ipfs.io/guides/con ...
+
+![Browing /ipns/ipns.io](/057-ipfs-desktop-0-9/browse-wild.png)
```
s/Browing/Browsing/

autonome commented on content/post/057-ipfs-desktop-0-9.md:
```diff
@@ -0,0 +1,65 @@
+
+### Use npm on ipfs
+
+In addition, we are introducing a new experiments section on the 'Settings' page which will allow you to enable new experimental features as they come in. To celebrate this new section, we are adding [npm on ipfs](https://github.com/ipfs-shipyard/npm-on-ipfs) to it! This is a project that allows you to use the `ipfs-npm` command to install your Node.js modules through the IPFS network!
```
Tough call... mention `ipfs-npm`, `npm-ipfs`, or both?!

autonome commented on content/post/057-ipfs-desktop-0-9.md:
```diff
@@ -0,0 +1,65 @@
+
+Adding `ipfs` command line tools to your system was actually introduced in 0.8.0, but it's always worth noting that you are just a click away of having the `ipfs` command available from your command line! This way, you don't have to worry about configuring your `PATH` variable or doing other confusing stuff to start using all IPFS capabilities. Just go to Settings and enable "IPFS command line to ...
+
+![IPFS command line tools](/057-ipfs-desktop-0-9/ipfs-cmd.png)
```
Not required at all, but would be neat if some of these images were gifs!

autonome commented on content/post/057-ipfs-desktop-0-9.md:
```diff
@@ -0,0 +1,65 @@
+
+Other notable features from previous releases include:
+
+- **Handle `ipfs://`, `ipns://` and `dweb:` links**. If you have IPFS Desktop installed and you click on [ipns://ipfs.io](ipns://ipfs.io), that request will go through be redirected to your own gateway via the app, or to the public gateway if yours is not online.
```
Just for extreme clarity, maybe add "... click on ipns://ipfs.io in your browser or any other application,..."

autonome commented on content/post/057-ipfs-desktop-0-9.md:
```diff
@@ -0,0 +1,65 @@
+Other notable features from previous releases include:
+
+- **Handle `ipfs://`, `ipns://` and `dweb:` links**. If you have IPFS Desktop installed and you click on [ipns://ipfs.io](ipns://ipfs.io), that request will go through be redirected to your own gateway via the app, or to the public gateway if yours is not online.
+- **Easily add files to IPFS**. Just drag and drop them to the application icon, either on the menubar for macOS users, of the shortcut icon for Windows users.
```
s/of/or/

autonome commented on content/post/057-ipfs-desktop-0-9.md:
```diff
@@ -0,0 +1,65 @@
+- **Handle `ipfs://`, `ipns://` and `dweb:` links**. If you have IPFS Desktop installed and you click on [ipns://ipfs.io](ipns://ipfs.io), that request will go through be redirected to your own gateway via the app, or to the public gateway if yours is not online.
+- **Easily add files to IPFS**. Just drag and drop them to the application icon, either on the menubar for macOS users, of the shortcut icon for Windows users.
+- **Add screenshots to IPFS**. You can click on 'Take Screenshot' on the app's menu or enable the global shortcut on "Settings". After taking a screenshot, a shareable link will be copied to your clipboard!
+- **Download any CID content**. If you would like to download the contents of some CID or IPFS/IPNS path, you can copy it, go to the dropdown menu and select 'Download Hash'. Then, just pick wherever you want to write those files to! This is also available through a global shortcut.
```
Can you note what the global shortcut is for each OS?

hacdias replied:
Hmmm... I wonder if it is really necessary to have on the blog post. I am saying this because these are features that are required to be manually enabled. Thus, the user will always see the shortcuts on settings

autonome commented on content/post/057-ipfs-desktop-0-9.md:
```diff
@@ -0,0 +1,65 @@
+
+We are now working on the [specification](https://github.com/ipfs-shipyard/cohosting/pull/2) of a new feature called cohosting, i.e., having a local copy of some website on your machine. You can do that just because you would like to keep a website active or because you want to read it offline later. Then, IPFS Companion or IPFS Desktop would re-check every 12 hours to make sure the website is up ...
+
+For the future of IPFS Desktop, we are now focusing much more on maintenance, bug fixing and improving the test suite to catch bugs before they go into production. Since this an Electron app, it is a bit harder to test than some other types of apps. If you're interesting in knowing more, please read the issue [#1121](https://github.com/ipfs-shipyard/ipfs-desktop/issues/1121).
```
"in knowing more or helping us build tests" 🤩

autonome commented on content/post/057-ipfs-desktop-0-9.md:
```diff
@@ -0,0 +1,65 @@
+
+# Don't forget...
+
+Don't forget to [install IPFS Desktop](https://github.com/ipfs-shipyard/ipfs-desktop/releases) today and be part of the decentralized Internet! We also recommend you to install our browser extension, [IPFS Companion](https://github.com/ipfs-shipyard/ipfs-companion/#install), which gives your browser superpowers. Both of this applications work well together. While IPFS Desktop makes your OS InterP ...
```
s/Both of this/Both of these/

terichadbourne commented:
@hacdias Just wanted to flag that you and I both used 057 for our post number in our drafts, so whoever gets merged second will need to make a minor update to change to 058. See #300.

hacdias commented:
As @lidel said, I don't mind converting all images to webp. We will get much smaller image files :)

hacdias commented:
I've just converted all images into jpegs for the sake of compatibility with browsers other than Firefox and Chrome. @autonome perhaps not including GIFs is the best thing to do also due to their size...

terichadbourne commented:
@hacdias I just submitted a commit updating your post number to 058 since I just published as 057. (Not sure which images are new and should have this change made and which are pulled from older posts.) 

I also updated your YAML to reflect today's date in the `date` and `URL` fields, but you'll need to adjust those again based on when you actually push this live. Maybe wait until tomorrow to avoid two posts on one day?

hacdias commented:
@terichadbourne thanks! Just updated the image directory and updated the date to tomorrow.

Tomorrow will be merging if I don't get any objections until then :)
